tools/topology/sof-apl-asrc-dmic.m4

```diff
@@ -0,0 +1,104 @@
+#
+# Topology for Apollo Lake with direct attach digital microphones array
+#
+
+# Capture configuration
+# DAI0 2ch 32b mic1-2
+# DAI1 2ch 32b mic1-2
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include Apollolake DSP configuration
+include(`platform/intel/bxt.m4')
+include(`platform/intel/dmic.m4')
+
+DEBUG_START
+
+#
+# Define the pipelines
+#
+# PCM6 <--- ASRC <--- DMIC6 (DMIC01)
+# PCM7 <--- ASRC <--- DMIC7 (DMIC16k)
```

Comment:
if you want to test the frequency changes, shouldn't you get an IPC to set the new destination rate?
It'd be useful for the host to be able to tweak the ASRC ratio (be it for testing or a a feature). For example if we record on DMIC and play on USB, you could have a loop in userspace that tracks the delta and tells the ASRC to adjust by producing fewer/more samples.

Reply:
This topology supports capture with variable rate 8 - 48 kHz (arecord -r x) from 48 kHz DAI and 16 kHz DAI. I'll at least try to update the max rate to 192 kHz to have more coverage for upsampling.

Your example sounds useful and interesting. What ALSA control type could pass the drift factor (single uint32_t Q2.30 value )? Timewise could we merge this first without that feature and add that later.